### PR TITLE
Survey templates

### PR DESCRIPTION
**Pull request by nickoki**
- Finished initial configuration of SurveyTemplatesController
- Finished Unit Tests for SurveyTemplatesController
- Updated SurveyTemplates views

### CODE REVIEW

File: app/controllers/survey_templates_controller.rb [hunk 1]
```diff
@@ -14,6 +17,82 @@ def index
...
+  #
+  def destroy
+    if @survey_template.destroy
+      flash[:success] = success_message(@survey_template, :delete)
+      redirect_to survey_templates_url, notice: 'Survey Template was successfully destroyed.'
+    else
+      flash[:error] = error_message(@survey_template, :delete)
+      redirect_back fallback_location: survey_templates_path
+    end
+  end
```
Comment (justdroo) [inline app/controllers/survey_templates_controller.rb]: To keep with the same pattern as #create, might want to extract this logic into a private method `destroy_survey_template` that checks: `@survey_template.destroyed?`

File: app/controllers/survey_templates_controller.rb [hunk 2]
```diff
@@ -14,6 +17,82 @@ def index
...
+  #
+  def update
+    if @survey_template.update(survey_template_params)
+      flash[:success] = success_message(@survey_template, :update)
+      redirect_to @survey_template, notice: 'Survey Template was successfully updated.'
+    else
+      flash.now[:error] = error_message(@survey_template, :update)
+      render :edit
+    end
+  end
```
Comment (justdroo) [inline app/controllers/survey_templates_controller.rb]: To keep with the same pattern as #create, might want to extract this logic into a private method `update_survey_template`

File: app/helpers/survey_templates_helper.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#
+# SurveyTemplatesHelper
+#
+module SurveyTemplatesHelper
+  #
+  # Renders controller action links for index table
+  #
+  def action_buttons(survey_template)
```
Comment (domarp-j) [inline app/helpers/survey_templates_helper.rb]: I'd recommend adding a quick test for this to make sure it functions properly.

File: test/controllers/survey_templates_controller_test.rb [hunk 4]
```diff
@@ -7,11 +7,73 @@ class SurveyTemplatesControllerTest < ActionController::TestCase
...
+
+      get(:show, params: { id: survey_template.id })
+      survey_template_instance = assigns(:survey_template)
+
+      assert_equal(survey_template_instance, survey_template, 'Did not return survey_templates')
+    end
+  end
+
+  describe '#create' do
+    it 'creates a new survey_template' do
```
Comment (domarp-j) [inline test/controllers/survey_templates_controller_test.rb]: I'd add a test for when invalid params are passed in (title is nil and/or name is nil) and make sure the create action *doesn't* create a new survey template.

File: test/controllers/survey_templates_controller_test.rb [hunk 5]
```diff
@@ -7,11 +7,73 @@ class SurveyTemplatesControllerTest < ActionController::TestCase
...
+
+    it 'redirects to sign-in page if a user is not signed in' do
+      post :create
+
+      assert_redirected_to new_user_session_path, 'Did not redirect to sign-in page'
+    end
+  end
+
+  describe '#update' do
+    it 'should allow admins to update survey_template' do
```
Comment (domarp-j) [inline test/controllers/survey_templates_controller_test.rb]: Similar to the create action - I'd add a test for when invalid params are passed in (title is nil and/or name is nil) and make sure the update action doesn't update the survey template.

File: app/views/survey_templates/show.html.erb [hunk 6]
```diff
@@ -1,10 +1,41 @@
...
-<%= @survey_template.title %>
-<%= @survey_template.name %>
+  <div class="row">
+    <div class="col-md-2">
+      <div class="alert alert-info">
+        <i>From this screen, you can view and edit your Survey Template: <strong>&quot;<%= @survey_template.name %>&quot;</strong>.</i>
+        <hr />
+        <i>Note that &quot;Survey Title&quot; represents what the End User will see when filling out this survey.</i>
+      </div>
+    </div>
```
Comment (justdroo) [inline app/views/survey_templates/show.html.erb]: This is awesome

File: app/controllers/survey_templates_controller.rb [hunk 7]
```diff
@@ -14,6 +17,82 @@ def index
...
+  #
+  def set_survey_template
     @survey_template = SurveyTemplate.find(params[:id])
   end
+
+  #
+  # Set and sanitize survey params
+  #
+  def survey_template_params
+    params.require(:survey_template).permit(:title, :name)
```
Comment (domarp-j) [inline app/controllers/survey_templates_controller.rb]: I'd add validations in survey_template.rb to make sure that every survey template has a name & title.

File: test/controllers/survey_templates_controller_test.rb [hunk 8]
```diff
@@ -7,11 +7,73 @@ class SurveyTemplatesControllerTest < ActionController::TestCase
...
+      admin = create :user, :admin
+      create :survey_template
+
+      sign_in admin
+
+      assert SurveyTemplate.first.valid?, 'SurveyTemplate was not created'
+    end
+
+    it 'redirects to sign-in page if a user is not signed in' do
+      post :create
```
Comment (domarp-j) [inline test/controllers/survey_templates_controller_test.rb]: After you add the validations to survey_template.rb, you'll have to pass in params here so that a survey template can actually be created. It'll probably look like `post(:create, params: { title: 'foo', name: 'bar' })`
